src/inet/http_server.c

```diff
@@ -199,7 +199,11 @@ find_route_callback(struct http_server_t *self_p,
     route_p = self_p->routes_p;
 
     while (route_p->path_p != NULL) {
-        if (strcmp(route_p->path_p, path_p) == 0) {
+        if (strstr(path_p, "?")) {
```

Comment:
Consider calling ``strstr()`` once before the while statement and save the returned value in a variable to reduc ethe number of calls to ``strstr()``.